synapse/lib/cell.py

```diff
@@ -1434,53 +1435,77 @@ async def initServiceNetwork(self):
             await self.addHttpsPort(port)
             logger.info(f'https listening: {port}')
 
+    async def _genAhaInfo(self):
+
+        # Default to static information
+        ahainfo = self.conf.get('aha:svcinfo')
+        if ahainfo is not None:
+            return ahainfo
+
+        # If we have not setup our dmon listener yet, do not generate ahainfo
+        if self.sockaddr is None:
+            return None
+
+        # If we have no dmon listener configured, do not generate ahainfo
+        turl = self.conf.get('dmon:listen')
+        if turl is None:
+            return None
+
+        # Dynamically generate the aha info based on config and runtime data.
+        urlinfo = s_telepath.chopurl(turl)
+
+        urlinfo.pop('host', None)
+        urlinfo['port'] = self.sockaddr[1]
+
+        celliden = self.getCellIden()
+        runiden = await self.getCellRunId()
+        ahalead = self.conf.get('aha:leader')
+
+        # If we are active, then we are ready by definition.
+        # If we are not active, then we have to check the nexsroot
+        # and see if the nexsroot is marked as ready or not. This
+        # status is set on mirrors when they have entered into the
+        # real-time change window.
+        if self.isactive:
+            ready = True
+        else:
+            ready = await self.nexsroot.isNexsReady()
```

Comment:
Do we have a mechanism in place to allow that ready status to be updated in aha when we enter/exit the real-time window?

Reply:
Lets discuss this further, since its a bit complicated.